coralproject/talk
# [CORL-2720] add configuration to premod emails with 3 or more periods

## What does this PR do?

Adds an option in `Admin > Config` that allows automatic pre-moderation of users with an email that has 3 or more periods.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

Adds the configuration options for `PremoderateEmailAddressConfig` to the `Settings` (tenant) type.

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- create a user with too many periods in their email
  - I used `too.many.periods@test.com`
- post a comment, see it is not pre-moderated
- sign in as your admin/mod user and toggle the premod filter on under `Admin > Configure > Moderation > Users > Email address`
- create another user with too many periods in their email
  - I used `too.many.periods.1@test.com`
- try and post a comment, see that you are pre-moderated
- using multi-site test, create SSO users that also have too many periods in their email
- attempt to post with these users, also see that they are pre-modded
- using your admin/mod user, remove the pre-mod status from one of these too many periods accounts
- sign in as that user and attempt to post a comment
- see that you can post the comment without being pre-moderated

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.
